crds/virtualdisksnapshot.yaml

```diff
@@ -0,0 +1,119 @@
+    singular: virtualdisksnapshot
+    kind: VirtualDiskSnapshot
+    shortNames:
+      - vdsnapshot
+      - vdsnapshots
+  preserveUnknownFields: false
+  versions:
+    - name: v1alpha2
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          description: |
+            Provides a resource for creating snapshots of existing virtual disks, which can be used as data sources for generating new virtual disks
+
+            A `VolumeSnapshot` is created under the hood of the resource.
+          required:
+            - spec
+          properties:
+            spec:
+              type: object
+              required:
+                - virtualDiskName
+                - volumeSnapshotClassName
+              properties:
+                virtualDiskName:
+                  type: string
+                  description: |
+                    The name of the virtual disk to take snapshot.
+                volumeSnapshotClassName:
+                  type: string
+                  description: |
+                    The name of the volume snapshot class to use while snapshotting virtual disk.
+                allowPotentiallyInconsistent:
+                  type: boolean
+                  default: false
+                  description: |
+                    Specify whether to allow the snapshotting of a virtual machine's disk that is attached to a running virtual machine without an agent, and as a result, cannot be frozen.
+
+                    If value is false, the snapshot of the virtual disk will be taken only in the following scenarios:
+                    - the virtual disk is not attached to any virtual machine.
+                    - the virtual disk is attached to a virtual machine that is powered off.
+                    - the virtual disk is attached to a virtual machine with an agent, and the freeze operation was successful.
+            status:
+              type: object
+              properties:
+                conditions:
+                  description: |
+                    The latest available observations of an object's current state.
+                  type: array
+                  items:
+                    type: object
+                    properties:
+                      lastProbeTime:
+                        description: Last time the condition was checked.
+                        format: date-time
+                        type: string
+                      lastTransitionTime:
+                        description: Last time the condition transit from one status to another.
+                        format: date-time
+                        type: string
+                      message:
+                        description: Human readable message indicating details about last transition.
+                        type: string
+                      reason:
+                        description: (brief) reason for the condition's last transition.
```

Comment:
That is mean 'brief'?

Reply:
Looks like it is used in the meaning of 'short description for the condition'. This is a common description for CRDs in accordance with other resources